pkg/doc/signature/ed25519signature2018/suite_test.go

```diff
@@ -42,37 +49,102 @@ func TestSignatureSuite_Sign(t *testing.T) {
 }
 
 func TestSignatureSuite_Verify(t *testing.T) {
-	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	ss := New(WithVerifier(&mockVerifier{}))
+
+	// happy path
+	err := ss.Verify("any key", []byte("any doc"), []byte("any signature"))
 	require.NoError(t, err)
-	require.NotEmpty(t, pubKey)
-	require.NotEmpty(t, privKey)
 
-	doc := []byte("hello world")
+	// no verifier defined
+	ss = New()
+	err = ss.Verify("any key", []byte("any doc"), []byte("any signature"))
+	require.Error(t, err)
+	require.Equal(t, ErrVerifierNotDefined, err)
 
-	signature := ed25519.Sign(privKey, doc)
-	require.NotEmpty(t, signature)
+	// verification error
+	ss = New(WithVerifier(&mockVerifier{verifyError: errors.New("verify error")}))
+	err = ss.Verify("any key", []byte("any doc"), []byte("any signature"))
+	require.Error(t, err)
+	require.EqualError(t, err, "verify error")
+}
 
-	ss := New()
-	publicKey := &verifier.PublicKey{Value: pubKey}
-
-	err = ss.Verify(publicKey, doc, signature)
-	require.Nil(t, err)
-
-	// test different message
-	err = ss.Verify(publicKey, []byte("different doc"), signature)
-	require.NotNil(t, err)
-	require.Contains(t, err.Error(), "signature doesn't match")
-
-	// test different signature
-	err = ss.Verify(publicKey, doc, []byte("signature"))
-	require.NotNil(t, err)
-	require.Contains(t, err.Error(), "signature doesn't match")
-
-	// test wrong public key size
-	publicKey.Value = []byte("key")
-	err = ss.Verify(publicKey, doc, signature)
-	require.NotNil(t, err)
-	require.Contains(t, err.Error(), "ed25519: bad public key length")
+func TestCryptoSignerAndVerifier(t *testing.T) {
+	lKMS := createKMS()
+	kid, kh := createKeyHandle(lKMS, kmsapi.Ed25519Type)
+
+	tinkCrypto, err := tinkcrypto.New()
+	require.NoError(t, err)
+
+	doc := []byte("test doc")
+
+	ss := New(WithSigner(NewCryptoSigner(tinkCrypto, kh)),
+		WithVerifier(NewLocalCryptoVerifier(tinkCrypto, lKMS)))
+	docSig, err := ss.Sign(doc)
+	require.NoError(t, err)
+	require.NotEmpty(t, docSig)
+
+	pubKH, err := kh.Public()
+	require.NoError(t, err)
+
+	// verify with key handle
+	err = ss.Verify(pubKH, doc, docSig)
+	require.NoError(t, err)
+
+	// verify with public key
+	pubKeyBytes, err := lKMS.ExportPubKeyBytes(kid)
+	require.NoError(t, err)
+
+	pubKey := &sverifier.PublicKey{
+		Type:  kmsapi.Ed25519Type,
+		Value: pubKeyBytes,
+	}
+
+	err = ss.Verify(pubKey, doc, docSig)
+	require.NoError(t, err)
```

Comment:
@Baha-sk the test is failing here saying that signature is invalid:
```
verify msg: verifier_factory: invalid signature
```
Do you have any idea why it's happening?

Reply:
Filed an issue regarding this: https://github.com/hyperledger/aries-framework-go/issues/1483
I disabled accepting a public key bytes input when verifying a signature via `LocalCryptoVerifier`.

Reply:
yeah, the signature has a special Tink prefix based on Tink key types, we can either fix it in Tink or manually remove the prefix from docSig and try `ss.Verify()` again... 

I will post an issue in Tink to see if we can get this added to Tink